internal/host/validator.go

```diff
@@ -674,50 +676,50 @@ func (v *validator) hasSufficientNetworkLatencyRequirementForRole(c *validationC
 		return ValidationSuccess
 	}
 
+	v.log.Debugf("Maximum network latency allowed for host %s: %f", c.host.ID, *c.clusterHostRequirements.Total.NetworkLatencyThresholdMs)
 	if len(c.host.Connectivity) == 0 {
 		return ValidationPending
 	}
-	latency, err := v.validateNetworkLatencyForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts)
-	if err != nil || len(latency) > 0 {
-		return ValidationFailure
-	}
-	return ValidationSuccess
+	status, _, _ := v.validateNetworkLatencyForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts, c.cluster.OpenshiftVersion, c.cluster.EmailDomain)
+	return status
 }
 
-func (v *validator) validateNetworkLatencyForRole(host *models.Host, clusterRoleReqs *models.ClusterHostRequirements, hosts []*models.Host) ([]string, error) {
+func (v *validator) validateNetworkLatencyForRole(host *models.Host, clusterRoleReqs *models.ClusterHostRequirements, hosts []*models.Host, openshiftVersion, emailDomain string) (ValidationStatus, []string, error) {
 	var connectivityReport *models.ConnectivityReport
 	connectivityReport, err := hostutil.UnmarshalConnectivityReport(host.Connectivity)
 	if err != nil {
-		return nil, fmt.Errorf("unable to unmarshall host connectivity for %s:%s", host.ID, err)
+		return ValidationFailure, nil, fmt.Errorf("unable to unmarshall host connectivity for %s:%s", host.ID, err)
 	}
 	failedHostIPs := map[string]struct{}{}
 	failedHostNames := []string{}
 	for _, r := range connectivityReport.RemoteHosts {
 		for _, l3 := range r.L3Connectivity {
+			hostname, role, err := GetHostnameAndRoleByIP(l3.RemoteIPAddress, hosts)
+			if err != nil {
+				v.log.Warnf("unable to determine host's role and hostname for IP %s: %s", l3.RemoteIPAddress, err)
+				return ValidationPending, nil, fmt.Errorf("unable to determine role and hostname for IP %s: %s", l3.RemoteIPAddress, err)
```

Comment:
I think it should be ValidationFailure and not ValidationPending

Reply:
Pending translates for "missing packet loss information", which I considered in the same scope as missing inventory data for other hosts.
When a host has not yet reported its inventory, does it show as pending of failure in the UI?

Reply:
Pending means that the user needs to do something and it's not the case here.
when host has not yet reported its inventory - we display it as failure, see: https://github.com/openshift/assisted-service/blob/master/internal/host/validator.go#L206

Reply:
ok, I'll change it to failure then